sound/soc/sof/ipc4-pcm.c

```diff
@@ -338,7 +343,12 @@ static int sof_ipc4_chain_dma_trigger(struct snd_sof_dev *sdev,
 	if (enable)
 		msg.primary |= SOF_IPC4_GLB_CHAIN_DMA_ENABLE_MASK;
 
-	return sof_ipc_tx_message_no_reply(sdev->ipc, &msg, 0);
+	ret = sof_ipc_tx_message_no_reply(sdev->ipc, &msg, 0);
+	/* Update the ChainDMA allocation state */
+	if (!ret && spcm->stream[direction].chain_dma_allocated != allocate)
```

Comment:
this check is redundant no? spcm->stream[direction].chain_dma_allocated != allocate

Reply:
We can set it unconditionally, yes. Pause/resume will cycle between RUNNING/PAUSE, but that's fine.